File: api/src/main/java/com/example/demo/model/Contato.java
package com.example.demo.model;

import lombok.Data;

import javax.persistence.*;

@Data
@Embeddable
public class Contato {

    @Column(name = "nome")
    private String nome;

    @Column(name = "email")
    private String email;

    @Column(name = "fone")
    private String fone;

    @Column(name = "cargo")
    private String cargo;
}

File: ui/src/app/model/contato.ts
import {FormControl, FormGroup, Validators} from '@angular/forms';

export class Contato {

  nome: string | undefined;
  email: string | undefined;
  fone: string | undefined;
  cargo: string | undefined;

  static criarFormulario(contato: Contato) {

    return new FormGroup({
      nome: new FormControl(contato.nome),
      email: new FormControl(contato.email,Validators.email),
      fone: new FormControl(contato.fone),
      cargo: new FormControl(contato.cargo)
    });
  }
}
